src/GovUk.Education.ExploreEducationStatistics.Publisher/PublisherHostBuilderExtensions.cs

```diff
@@ -0,0 +1,193 @@
+                    .AddScoped<GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces.IMethodologyService, GovUk.Education.ExploreEducationStatistics.Content.Services.MethodologyService>()
+                    .AddScoped<IMethodologyCacheService, MethodologyCacheService>()
+                    .AddScoped<IPublicationService, PublicationService>()
+                    .AddScoped<IPublicationCacheService, PublicationCacheService>()
+                    .AddScoped<GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces.IReleaseService, GovUk.Education.ExploreEducationStatistics.Content.Services.ReleaseService>()
+                    .AddScoped<IReleaseFileRepository, ReleaseFileRepository>()
+                    .AddScoped<IReleaseCacheService, ReleaseCacheService>();
+
+                // Only set up the `PublicDataDbContext` in non-integration test
+                // environments. Otherwise, the connection string will be null and
+                // cause the data source builder to throw a host exception.
+                if (!hostEnvironment.IsIntegrationTest())
+                {
+                    var connectionString = configuration.GetConnectionString("PublicDataDb")!;
+
+                    if (hostEnvironment.IsDevelopment())
+                    {
+                        var dataSourceBuilder = new NpgsqlDataSourceBuilder(connectionString);
+
+                        // Set up the data source outside the `AddDbContext` action as this
+                        // prevents `ManyServiceProvidersCreatedWarning` warnings due to EF
+                        // creating over 20 `IServiceProvider` instances.
+                        var dbDataSource = dataSourceBuilder.Build();
+
+                        services.AddDbContext<PublicDataDbContext>(options =>
+                        {
+                            options
+                                .UseNpgsql(dbDataSource)
+                                .EnableSensitiveDataLogging();
+                        });
+                    }
+                    else
+                    {
+                        services.AddDbContext<PublicDataDbContext>(options =>
+                        {
+                            var sqlServerTokenProvider = new DefaultAzureCredential();
+                            var accessToken = sqlServerTokenProvider.GetToken(
+                                new TokenRequestContext(scopes: new[]
+                                {
+                                    "https://ossrdbms-aad.database.windows.net/.default"
+                                })).Token;
+
+                            var connectionStringWithAccessToken =
+                                connectionString.Replace("[access_token]", accessToken);
+
+                            var dbDataSource = new NpgsqlDataSourceBuilder(connectionStringWithAccessToken).Build();
+
+                            options.UseNpgsql(dbDataSource);
+                        });
```

Comment:
Shouldn't we put in the same check that we added to Admin to only do this if the public data db exists in the environment yet? Otherwise I think there will be an issue if this is deployed to an Azure environment without it.

Reply:
Yup good spot thanks.  I've put it in, fed it as a config value from the ARM template using the existing param you put in, and added a NoOp implementation of the service which relies on the DbContext being available.